ext/drsyms/libelftc/windows-mingw-port-common.patch

```diff
@@ -0,0 +1,1779 @@
+diff --git  common/_elftc.h common/_elftc.h
+index 5be8145b..898bee40 100644
+---  common/_elftc.h
++++ common/_elftc.h
+@@ -325,2 +325,6 @@ struct name {							\
+ 
++#ifndef ELFTC_VCSID
++#define ELFTC_VCSID(ID) /* nothing */
++#endif
++
+ #endif	/* ELFTC_VCSID */
+@@ -372,2 +376,10 @@ extern const char *__progname;
+ 
++#if defined _WIN32 && ! defined __CYGWIN__
++#include "windows-mingw/endian.h"
++
++#define	ELFTC_BYTE_ORDER			__BYTE_ORDER
++#define	ELFTC_BYTE_ORDER_LITTLE_ENDIAN		__LITTLE_ENDIAN
++#define	ELFTC_BYTE_ORDER_BIG_ENDIAN		__BIG_ENDIAN
++#endif
++
+ #if defined(__APPLE__)
+@@ -449,3 +461,7 @@ extern const char *__progname;
+ 
++#if !(defined _WIN32 && ! defined __CYGWIN__)
+ #include <sys/param.h>
++#else
++#include "windows-mingw/sys_param.h"
++#endif
+ #include <sys/endian.h>
+@@ -468,3 +484,7 @@ extern const char *__progname;
+ 
++#if !(defined _WIN32 && ! defined __CYGWIN__)
+ #include <sys/param.h>
++#else
++#include "windows-mingw/sys_param.h"
++#endif
+ #include <sys/endian.h>
+diff --git  common/windows-mingw/_ansi.h common/windows-mingw/_ansi.h
+new file mode 100644
+index 00000000..3a8d0838
+--- /dev/null
++++ common/windows-mingw/_ansi.h
+@@ -0,0 +1,73 @@
++/* Provide support for both ANSI and non-ANSI environments.  */
```

Comment:
This looks like a copy of a real header?  What about its copyright and license?

It looks like there are many copies of real headers, with different copyrights and licenses, below.  In fact it looks like that is the source of the unusual "sys_xxxx.h" names?  If we include copies of other headers we have to be sure to satisfy their licenses.  But if all of these are present on a mingw system, do we need to make copies?  We can just point at the ones installed for mingw, right?  We can just use `-I` or `-isystem` or other include flags to find the mingw versions?  Then we don't need all of this in the patch file.

Reply:
These headers went from Cygwin project.

Reply:
Like I said, there are license and copyright issues with just copying files.  It would be simplest to require the developer to have these on the machine and to point at them, rather than copying them here, if they are part of standard installations.